#include "mainwindow.h"
#include "ui_mainwindow.h"

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    QSslConfiguration sslConf = QSslConfiguration::defaultConfiguration();
    sslConf.setPeerVerifyMode(QSslSocket::VerifyNone);
    QSslConfiguration::setDefaultConfiguration(sslConf);

    htmlView = nullptr;
    ui->setupUi(this);

    ui->progressBar->setRange(0,65535);
}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow::loadFinished(bool x)
{
    webPage->toHtml([&](const QString& result)
    {
        if (result.contains("bea.com"))
        {
            ui->progressBar->setValue(ui->spinBox->value());

            if (result.contains("Received a response"))
            {
                qDebug() << QString("%1:%2 found open").arg(ui->lineEdit_scanhost->text()).arg(ui->spinBox->value());
                ui->plainTextEdit->appendPlainText(QString("%1:%2 found open").arg(ui->lineEdit_scanhost->text()).arg(ui->spinBox->value()));
            }
            delete webPage;
            if (ui->spinBox->value() == 65535) return;
            ui->spinBox->setValue(ui->spinBox->value()+1);
            on_pushButton_clicked();
        }
    }
    );
}

void MainWindow::showEvent(QShowEvent *event)
{
    htmlView = new QWebEngineView(ui->frame);
    htmlView->load(QUrl("about:blank"));

    QFile disclaimerHTML( ":/htmls/in.html" );
    disclaimerHTML.open( QFile::ReadOnly );
    QString sDisclaimerHTML( disclaimerHTML.readAll() );

    htmlView->setHtml(sDisclaimerHTML);
    htmlView->show();
    htmlView->resize(ui->frame->size());
}

void MainWindow::resizeEvent(QResizeEvent *event)
{
    if (htmlView) htmlView->resize(ui->frame->size());
}

void MainWindow::on_splitter_splitterMoved(int pos, int index)
{
    if (htmlView) htmlView->resize(ui->frame->size());
}

void MainWindow::on_pushButton_clicked()
{
    webPage = new WebPage(this);
    QString url = QString("%1?operator=http://%2:%3/&rdoSearch=name&txtSearchname=sdf&txtSearchkey=&txtSearchfor=&selfor=Business+location&btnSubmit=Search").arg(ui->lineEdit_url->text()).arg(ui->lineEdit_scanhost->text()).arg(ui->spinBox->value());
//    qDebug() << "Loading url" << url;
    webPage->load(QUrl( url));
    htmlView->setPage(webPage);
    connect(webPage, &QWebEnginePage::loadFinished, this, &MainWindow::loadFinished);
}
